src/Audio/Output/PlayerAudioSource.cs

```diff
@@ -98,38 +103,57 @@ public void UpdatePlayer()
             // If the player is on the other side of something to the listener, then the player's voice should be muffled
             bool mufflingEnabled = ClientSettings.Muffling;
             float wallThickness = LocationUtils.GetWallThickness(capi, player, capi.World.Player);
+            float wallThicknessWeighting = WorldConfig.GetFloat("wallThicknessWeighting");
             if (capi.World.Player.Entity.Swimming)
                 wallThickness += 1.0f;
 
             lowpassFilter?.Stop();
             if (mufflingEnabled && wallThickness != 0)
             {
-                lowpassFilter = lowpassFilter ?? new FilterLowpass(source);
+                lowpassFilter = lowpassFilter ?? new LowpassFilter(source);
                 lowpassFilter.Start();
-                lowpassFilter.SetHFGain(Math.Max(1.0f - (wallThickness / 2), 0.1f));
+                lowpassFilter.SetHFGain(Math.Max(1.0f - (wallThickness / wallThicknessWeighting), 0.1f));
             }
 
+            bool toBeImplementedToggle = false;
+            // DEACTIVATED : TO BE IMPLEMENTED
             // If the player is in a reverberated area, then the player's voice should be reverberated
-            bool isReverberated = false;
-            if (isReverberated)
+            reverbEffect?.Stop();
+            if (toBeImplementedToggle)
             {
-
+                BlockPos plrpos = new BlockPos();
+                plrpos.Set((int)speakerPos.X, (int)speakerPos.Y, (int)speakerPos.Z);
+                Room room = capi.ModLoader.GetModSystem<RoomRegistry>().GetRoomForPosition(plrpos);
+                // Check whether it is a proper room, or something like a room i.e. with a roof, for example a natural cave
+                bool inEnclosedRoom = room.ExitCount == 0 || room.SkylightCount < room.NonSkylightCount;
+                if (inEnclosedRoom) 
+                {
+                    reverbEffect = reverbEffect ?? new ReverbEffect(source);
+                    reverbEffect.Start();
+                }
```

Comment:
Implementation details should be abstracted away to utility function or some delegated class, PlayerAudioSource is a high-level class that shouldn't know nor handle any of this